src/ssl.c

```diff
@@ -6963,9 +6984,14 @@ long wolfSSL_CTX_set_session_cache_mode(WOLFSSL_CTX* ctx, long mode)
     if (mode == SSL_SESS_CACHE_OFF)
         ctx->sessionCacheOff = 1;
 
-    if (mode == SSL_SESS_CACHE_NO_AUTO_CLEAR)
+    if ((mode & SSL_SESS_CACHE_NO_AUTO_CLEAR) != 0)
```

Comment:
What about the case that mode = 0x0c and the flag SSL_SESS_CACHE_NO_AUTO_CLEAR = 0x09. This would result in taking a true branch when the mode was not set.